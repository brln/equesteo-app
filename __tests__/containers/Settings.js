import configureStore from 'redux-mock-store'
import React from "react"
import {fromJS, Map} from "immutable"
import renderer from "react-test-renderer"
import { Provider } from 'react-redux'
import thunk from "redux-thunk"

import SettingsContainer from '../../containers/Settings'


const mockStore = configureStore([thunk])

test('render', () => {
  const userID = 'userID1'
  const store = mockStore(fromJS({
    localState: {
      userID: userID
    },
    pouchRecords: {
      users: {
        userID1: {
          _id: userID
        }
      }
    },
  }))

  const tree = renderer.create(
    <Provider store={store}>
      <SettingsContainer />
    </Provider>).toJSON()
  expect(tree).toMatchSnapshot()
})

