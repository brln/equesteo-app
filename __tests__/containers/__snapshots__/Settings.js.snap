// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="always"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          header={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": null,
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#4b9ec1",
              }
            }
          >
            Privacy:
          </Text>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "marginBottom": 20,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 20,
                }
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 31,
                    "width": 51,
                  }
                }
                value={false}
              />
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Default my rides to publicly viewable.
              </Text>
            </View>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "marginBottom": 20,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 20,
                }
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 31,
                    "width": 51,
                  }
                }
                value={false}
              />
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Opt-out of leaderboards.
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          }
        }
      >
        <View
          header={true}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderBottomWidth": null,
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#4b9ec1",
              }
            }
          >
            Sounds:
          </Text>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "marginBottom": 20,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 20,
                }
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 31,
                    "width": 51,
                  }
                }
                value={false}
              />
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Play distance alerts.
              </Text>
            </View>
          </View>
        </View>
        <View
          cardBody={true}
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": -5,
                "paddingLeft": -5,
                "paddingRight": -5,
                "paddingTop": -5,
              },
              Object {
                "marginBottom": 20,
                "marginLeft": 20,
                "marginRight": 20,
              },
            ]
          }
        >
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "row",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "marginRight": 20,
                }
              }
            >
              <RCTSwitch
                accessibilityRole="button"
                onChange={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "height": 31,
                    "width": 51,
                  }
                }
                value={false}
              />
            </View>
            <View
              style={
                Object {
                  "flex": 6,
                  "justifyContent": "center",
                }
              }
            >
              <Text>
                Disable gps alerts.
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
